backend/src/main/java/com/zeros/backend/service/RouteServiceImpl.java

```diff
@@ -0,0 +1,45 @@
+package com.zeros.backend.service;
+
+import java.math.BigDecimal;
+import java.time.Duration;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.google.maps.DistanceMatrixApi;
+import com.google.maps.GeoApiContext;
+import com.google.maps.model.DistanceMatrix;
+import com.google.maps.model.TravelMode;
+
+@Service
+public class RouteServiceImpl implements RouteService{
+
+	private final GeoApiContext geoApiContext;
+
+	@Autowired
+	public RouteServiceImpl(final GeoApiContext geoApiContext) {
+		this.geoApiContext = geoApiContext;
+	}
+
+	@Override
+	public Route estimateRoute(Location userlocation, Location restauranteLocation
+			, TravelMode travelMode) {
```

Comment:
I preferred that class is generic . We give it a location a and location b and it calculate the route

Reply:
do you mean params name ???

Reply:
ItineraryService ???

Reply:
About the function parms. 
This service has no idea that you search a restaurant.
You give him a departure location and destination and he do his work